sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -3,209 +3,253 @@
-            )
-            for config in configurations:
+    provider._trim_prefixes = sorted(kwargs.pop("trimmed_key_prefixes", []), key=len, reverse=True)
 
-                trimmed_key = config.key
-                # Trim the key if it starts with one of the prefixes provided
-                for trim in provider._trim_prefixes:
-                    if config.key.startswith(trim):
-                        trimmed_key = config.key[len(trim) :]
-                        break
+    secret_clients = key_vault_options.secret_clients if key_vault_options else {}
 
-                if config.content_type == KEY_VAULT_REFERENCE_CONTENT_TYPE:
-                    secret = provider.__resolve_keyvault_reference(config, key_vault_options, secret_clients)
-                    provider._dict[trimmed_key] = secret
-                elif provider.__is_json_content_type(config.content_type):
-                    try:
-                        j_object = json.loads(config.value)
-                        provider._dict[trimmed_key] = j_object
-                    except json.JSONDecodeError:
-                        # If the value is not a valid JSON, treat it like regular string value
-                        provider._dict[trimmed_key] = config.value
-                else:
+    for select in selects:
+        configurations = provider._client.list_configuration_settings(
+            key_filter=select.key_filter, label_filter=select.label_filter
+        )
+        for config in configurations:
+
+            trimmed_key = config.key
+            # Trim the key if it starts with one of the prefixes provided
+            for trim in provider._trim_prefixes:
+                if config.key.startswith(trim):
+                    trimmed_key = config.key[len(trim) :]
+                    break
+
+            if config.content_type == KEY_VAULT_REFERENCE_CONTENT_TYPE:
+                secret = __resolve_keyvault_reference(config, key_vault_options, secret_clients)
+                provider._dict[trimmed_key] = secret
+            elif __is_json_content_type(config.content_type):
+                try:
+                    j_object = json.loads(config.value)
+                    provider._dict[trimmed_key] = j_object
+                except json.JSONDecodeError:
+                    # If the value is not a valid JSON, treat it like regular string value
                     provider._dict[trimmed_key] = config.value
+            else:
+                provider._dict[trimmed_key] = config.value
+    return provider
+
+def __buildprovider(connection_string:str, endpoint:str, credential,
+        key_vault_options:AzureAppConfigurationKeyVaultOptions):
+    provider = AzureAppConfigurationProvider()
+    headers = {}
+    correlation_context = "RequestType=Startup"
+
+    if key_vault_options and (
+        key_vault_options.credential or key_vault_options.secret_clients or key_vault_options.secret_resolver
+    ):
+        correlation_context += ",UsesKeyVault"
```

Comment:
So we are adding RequestType and UsesKeyVault tracing right now. Is there a way to get host type and dev environment tracing for python apps?

Reply:
It was already there. This was just moved. I can work on adding those, but not in this branch.